asv_benchmarks/benchmarks/datasets.py

```diff
@@ -0,0 +1,146 @@
+import os
+
+import numpy as np
+import scipy.sparse as sp
+from joblib import Memory
+
+from sklearn.decomposition import TruncatedSVD
+from sklearn.datasets import (make_blobs, fetch_20newsgroups,
+                              fetch_openml, load_digits, make_regression,
+                              make_classification, fetch_olivetti_faces)
+from sklearn.preprocessing import MaxAbsScaler, StandardScaler
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.model_selection import train_test_split
+
+# memory location for caching datasets
+M = Memory(location=os.path.dirname(os.path.realpath(__file__)) + "/cache")
```

Comment:
Can't we reuse the scikit-learn datasets  dir (maybe with a specific "asv-cache"" folder there)?

Reply:
I use the cache folder for the datasets but also for temporary stuff. I think it makes more sense to keep everything in a same place. And also to keep everything related to the benchmarks in the benchmarks folder